docs/types/mapping-types.rst

```diff
@@ -52,6 +58,34 @@ each ``_KeyType``, recursively. For example with a mapping:
         }
     }
 
+The example below uses a mapping type inside another mapping type. The important
+point to remember are that the mapping inside the mapping is not the return value,
```

Comment:
That should be ``The important point to remember is`` not ``are``... but also: "the return value" of what? I'm not sure I understand the purpose of this sentence :-).

Reply:
Corrected that. The point is to clarify that the returned value isn't the mapping inside the mapping, but the value inside the nested mapping. That's what I understood from you anyway?

Reply:
When you're talking about "the returned value", you have to say what you are talking about that returns something - the index access of the outer mapping? The index access of the outer mapping in fact *does* return "the mapping inside the mapping" - only the second index access yields the value contained in the inner mapping...

```
contract C {
    mapping(address => mapping(address => bool)) public allowances;
    function f(address X, address Y) public view {
        mapping(address => bool) storage allowancesForX = allowances[X]; // first index access is a mapping
        bool allowanceForXandY = allowances[X][Y]; // two index accesses give the contained value
        bool allowanceForXandY2 = allowancesForX[Y]; // which is the same as accessing the contained mapping at Y
        assert(allowanceForXandY == allowanceForXandY2);
    }
}

```